### ISSUE TITLE
how

### ISSUE THREAD

**Issue by dregypt** [hasan29351/Django-Blog#14]
nice app how add on my project my users not models it by view
dr and patient.  user.profile types

**Comment by hasan-py** [comment]
I don't understand what problem you face. Please explain it in details.

**Comment by dregypt** [comment]
could talk telegram   what's your number

**Comment by dregypt** [comment]
need your advice this app to select by autocomplte doctor not work


class Profile(models.Model):
    user = models.OneToOneField(User, on_delete=models.CASCADE)
    user_type = models.CharField(max_length=15, null=True,blank=False)
    address = models.CharField(max_length=256, null=True, blank=True)
    phone_regex = RegexValidator(regex=r'^[6-9]\d{9}$', message="Please enter valid mobile number.")
    mobile = models.CharField(validators=[phone_regex], max_length=10, blank=True)
    blood_group = models.IntegerField(choices=BLOOD_GROUP_CHOICE, default=1)
    gender = models.IntegerField(choices=GENDER_CHOICE, default=1)
    age = models.PositiveIntegerField(default=0)



class AppointmentForm(forms.Form):# BSModalModelForm
    #doctor_name= Profile.objects.all()#filter(user_type__in= ('Doctor'))#.filter(dod=False).count()


    #doctor_name= Profile.objects.filter(user_type__in= ('Doctor'))#.filter(dod=False).count()


    #doctor_name = forms.ModelMultipleChoiceField(queryset=Profile.objects.filter(user_type='Doctor'), widget=Select2MultipleWidget, required=True)
    #doctor_name = forms.ModelChoiceField(queryset=Profile.objects.filter(user_type='Doctor'))

    app_date = forms.DateField(widget=forms.TextInput(attrs={'placeholder':'MM/DD/YYYY', 'class': 'form-control', 'id': 'datepicker1', 'autocomplete': 'off'}))

    #class Meta:
        #model=Appointment
        #fields = '__all__'



    def clean(self):
        super(AppointmentForm, self).clean()

        appointment_date = self.cleaned_data.get('app_date')

        if appointment_date <= date.today() + timedelta(days=1):
            self.add_error('app_date', 'Please select valid date for appointment.')

        if appointment_date > date.today() + timedelta(days=15):
            self.add_error('app_date', 'Appointment date must be between 15 days.')

        return self.cleaned_data


        widgets = {
            "full_name": DoctorWidget(select2_options={'width': '100%', }),
        }



def new_appointment(request):
    if request.user.is_authenticated:
        if request.session['user_type'] == 'Receptionist':
            user = request.user
            booked_by = User.objects.get(pk=user.id)#i add
            if request.method == 'POST':
                patient_username = request.POST['username']
                doc_username = request.POST['doc_username']
                #recp_username = request.POST['recp_username']
                app_date = datetime.datetime.strptime(request.POST['date'], "%Y-%m-%d").date()
                patient_data = User.objects.filter(username=patient_username)
                doctor_data = User.objects.filter(username=doc_username)
                #recp_data = User.objects.filter(username=recp_username)
                today = date.today()
                if app_date >= today:
                    for i in patient_data:
                        full_name = i.first_name +" "+ i.last_name
                        mobile = i.profile.mobile
                        file_number = i.profile.file_number

                    for i in doctor_data:
                        doctor_name = i.first_name + " " + i.last_name
                        dep = i.profile.department

                    #for i in recp_data:
                        #recp_name = i.first_name + " " + i.last_name
                        #dep = i.profile.department


                    app_count = Appointment.objects.filter(doctor_name=doctor_name, app_date=app_date).count()
                    if app_count < 10:
                        app_no = app_count + 1
                        check_data = Appointment.objects.filter(user=patient_username, doctor_name=doctor_name, app_date=app_date)
                        if check_data:
                            messages.warning(request, f'Appointment is already created for <strong>{full_name}</strong> on date <strong>{app_date}</strong>.')

                        else:
                            new_app = Appointment(user=patient_username, app_date=app_date, doctor_name=doctor_name, department=dep, book_date=today, full_name=full_name, mobile=mobile,file_number=file_number, appointment_no=app_no,booked_by=booked_by)
                            new_app.save()
                            messages.success(request, f'New appointment is created for <strong>{full_name}</strong> on date <strong>{app_date}</strong>.')

                    else:
                        messages.success(request, f'Appointment is not available on date {app_date}, please select another date for appointment.')
                else:
                    messages.warning(request, f'New appointment date must be greater than or equal to today.')
                return redirect('/arecp/lis/all')#/receptionist/patient_list/
        else:
            return redirect('/')
    else:
        return redirect('/')




{% extends 'accounts/receptionist/base.html' %}
{% load static %}
{% block title %} Patient List {% endblock %}

{% block page_location %}Patient List قائمة المرضي المسجلين <a href="{% url 'create_patientHistory' %}"  title="Add field"><img src="http://127.0.0.1:8000/media/images/patientrecord.jpg" style="width: 33px; height: 25px"/> {% endblock %}


<style media="screen">
  .ui-front {
      z-index: 9999;
  }
</style>
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/select2@4.1.0-rc.0/dist/css/select2.min.css">
 <script src="https://cdn.jsdelivr.net/npm/select2@4.1.0-rc.0/dist/js/select2.min.js"></script>  
  <script type="text/javascript" src="http://jsearchdropdown.sourceforge.net/jquery-1.7.1.min.js"></script>
      <script type="text/javascript" src="http://jsearchdropdown.sourceforge.net//jquery.searchabledropdown.js"></script>
     <script type="text/javascript">
       $(document).ready(function() {
    $('#docname').searchable();

    $('#docname').change(function() {
      $(this).autocomplete('search', '');
    });
  });
    </script>
  

      <script src="{% static 'js/jQuery.js' %}"></script>



      {{ form.media.css }}










<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script>




<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/select2@4.1.0-rc.0/dist/css/select2.min.css" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/select2-bootstrap-5-theme@1.3.0/dist/select2-bootstrap-5-theme.min.css" />

<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/select2-bootstrap-5-theme@1.3.0/dist/select2-bootstrap-5-theme.rtl.min.css" />


<script src="https://cdn.jsdelivr.net/npm/jquery@3.5.0/dist/jquery.slim.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/bootstrap@5.1.3/dist/js/bootstrap.bundle.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/select2@4.1.0-rc.0/dist/js/select2.min.js"></script>












{% block main_content %}


<div class="col-lg-8 h-100 mx-auto shadow-sm p-3">

    <form action="/receptionist/patient_list/" method="post" class="mb-4">
        {% csrf_token %}
        <div class="row g-3 justify-content-center">
            <div class="col-lg-3">
                <select class="form-select" name="search_by" aria-label="Default select example" required>
                    <option value="" selected>Search By</option>
                    <option value="All">All</option>
                    <option value="Name">Name</option>
                    <option value="Mobile">Mobile Number</option>
                </select>
            </div>
            <div class="col-lg-4">
                <input type="text" name="get_text" id="get_text" class="form-control"
                    placeholder="Type name or mobile number" autocomplete="off">
            </div>
            <button type="submit" class="btn btn-primary col-lg-3">Search</button>
        </div>
    </form>

    <p class="text-secondary text-center pb-3 border-bottom fs-5 fw-bold">Patient List</p>

    <table class="table table-light table-bordered" id="patient" width='100%'>
        <thead class="table-dark text-center align-middle">
            <tr>
                <th>Sr._No.</th>
                <th>Patient_Name</th>
                <th>File_Number</th>
                <th>QR</th>
                <th>Mobile_Number</th>
                <th>Address</th>
                <th>registerd by</th>
                <th>full_details</th>
                <th>Medical_Records</th>

                <th>Appointment_writeusername</th>



            </tr>
        </thead>
        <tbody class="text-center align-middle">
            {% for field in object_list %}
            {% if field.user_type == "Patient" %}

            <tr>
                <td>{{forloop.counter}}</td>
                <td>{{field.user.get_full_name}}</td>

                <td>{{field.file_number}}</td>
                <td class="col-md-1"><img id="IMG" src="{{field.qr_code.url}}" style="width:22px; height: 22px" /></td>
                <td>{{field.mobile}}</td>
                <td>{{field.address}}</td>
                <td>{{field.register_by}}</td>
                <td> <a href="{% url 'patient-detail' field.id %}" class="btn btn-outline-white-secondary">
                      <i class="fa fa-edit"></i> Details
                  </a></td>
                <td><a href="/receptionist/viewrecord/patient={{field.user.username}}"
                        class="btn btn-warning text-center"><i class="fas fa-clipboard me-2"></i><span>View</span></td>
                <td><a type="button" class="btn btn-warning text-center" data-bs-toggle="modal"
                        data-bs-target="#appointment{{forloop.counter}}"><i
                            class="far fa-calendar-plus me-2"></i><span>Create</span></td>





                
                <div class="modal fade" id="appointment{{forloop.counter}}" tabindex="-1"
                    aria-labelledby="exampleModalLabel" aria-hidden="true">
                    <div class="modal-dialog ui-front">
                        <div class="modal-content">
                            <div class="modal-header">
                                <h5 class="modal-title" id="exampleModalLabel">New Appointment</h5>
                                <button type="button" class="btn-close" data-bs-dismiss="modal"
                                    aria-label="Close"></button>
                            </div>
                            <div class="modal-body">
                              <div class="modal-body">
                                  <p class="text-center fs-6 fw-bold">Do you want to create new appointment for
                                      '{{field.user.get_full_name}}'</p>
                                  <form action="/receptionist/new_appointment/" method="post">
                                      {% csrf_token %}
                                      <input type="hidden" name="username" value="{{field.user.username}}">
                                      <div class="row justify-content-center">
                                          <div class="col-sm-5">
                                              <select class="form-select" name="doc_username"
                                                  aria-label="Default select example" required>
                                                  <option value="" selected>Select Doctor Name</option>
                                                  {% for doc in doctors %}
                                                  <option value="{{doc.user.username}}">{{doc.user.get_full_name}}
                                                  </option>
                                                  {% endfor %}
                                              </select>
                                          </div>

                                        <div class="col-sm-5">
                                            <input type="date" name="date" id="date" class="form-control" required>
                                        </div>
                                    </div>
                            </div>

                            <div class="modal-footer">
                                <button type="submit" class="btn btn-success">Submit</button>
                                <button type="button" class="btn btn-secondary" data-bs-dismiss="modal">Close</button>
                            </div>
                            </form>
                        </div><


                    </div>
                </div>

                <td> <a href="{% url 'add_appointmentnew' field.id %}" class="btn btn-outline-white-secondary">
                      <i class="fa fa-edit"></i>Appointment
                  </a></td>
            </tr>

            {% endif %}
            {% endfor %}
        </tbody>
    </table>
</div>

<script>
    $(document).ready(function () {
        $('#patient').DataTable({
            "dom": '<"top">rt<"bottom"p><"clear">',
            "ordering": false,
            "scrollX": true,
            "pageLength": 5,
        });
    });
</script>

<script src="https://code.jquery.com/jquery-1.12.4.js"></script>
<script src="https://code.jquery.com/ui/1.12.1/jquery-ui.js"></script>
{% comment %} <script>
    $( function() {

    $( "#id_docc" ).autocomplete({
      source: '{% url "dr-auto" %}'
    });
  } );
</script>
{% endcomment %}


<script src="https://code.jquery.com/jquery-1.12.4.js"></script>
<script src="https://code.jquery.com/ui/1.12.1/jquery-ui.js"></script>
{% comment %} <script>


    $( function() {

    $( ".form-select" ).autocomplete({
      source: '{% url "dr-auoto" %}'
    });
  } );
</script>

<script type="text/javascript">
//$('select').val('option-value');
//$("#docname").searchable();

  //$('select').val('option-value');
  //$("#docname").searchable();
  $(document).ready(function() {
  // Insere classe no primeiro item de produto
  $('#id_doctor_name').addClass('clProdutotooo');

  $('.textinput textInput').select2()
});
</script>
<script>
 $(document).ready(function() {
 console.log('');
 $('.textinput textInput').select2();
 });
</script>

<script type="text/javascript">


$(document).ready(function(){
    $('#id_data').hide();
})
$('#id_data').parents('p:first').hide();

</script>


{% endcomment %}



{% endblock %}

**Comment by dregypt** [comment]
class Appointment(models.Model):

    class Meta:
        unique_together = ('doctor_name', 'app_date', 'timeslot')

    TIMESLOT_LISTS = (
        (0, '09:00 – 09:30'),
        (1, '10:00 – 10:30'),
        (2, '11:00 – 11:30'),
        (3, '12:00 – 12:30'),
        (4, '13:00 – 13:30'),
        (5, '14:00 – 14:30'),
        (6, '15:00 – 15:30'),
        (7, '16:00 – 16:30'),
        (8, '17:00 – 17:30'),
    )


    #patient = models.ForeignKey(User,related_name="appointmentcos",on_delete=models.CASCADE,blank=True,null=True)
    #dermatologist = models.ForeignKey(User,related_name="appointmentcoss" ,on_delete=models.CASCADE,blank=True,null=True)
    #Aesthetic_session = models.CharField(max_length=100,choices=Aesthticc_Types,blank=True,null=True)
    #procedure = models.CharField(max_length=100,choices=Procedure_Types,blank=True,null=True)
    #Laser_Sessions = models.CharField(max_length=100,choices=Laser_Types,blank=True,null=True)
    #date = models.DateField(help_text="YYYY-MM-DD")
    timeslot = models.IntegerField(max_length=100,choices=TIMESLOT_LISTS,blank=True,null=True)
    appoint_type = models.CharField(max_length=1010,blank=True,null=True)


    # id = models.UUIDField(primary_key = True, default = uuid.uuid4, editable = False)
    user = models.CharField(max_length=256, null=False, blank=False)
    full_name = models.CharField(max_length=256, null=False, blank=False, default='')
    mobile = models.CharField(max_length=10, null=False, blank=False, default='')
    file_number = models.CharField(max_length=14, null=False, blank=False, default='')
    app_date = models.DateField(null=False, blank=False)
    book_date = models.DateField(null=False, blank=False, auto_now=True)
    doctor_name = models.CharField(max_length=256, null=False, blank=False)
    department = models.CharField(max_length=100, null=False, blank=False

**Comment by ULis3h** [comment]
Hello,

We have an exciting opportunity for you! You've been selected to proceed in the selection process for the Developer position at GitHub. Congratulations on your achievement!

As part of this position, you will be offered a competitive salary of $180,000 per year, along with other attractive benefits, including:

- Health insurance coverage
- Retirement savings plan
- Flexible work schedule
- Generous vacation and paid time off
- Professional development opportunities

To proceed with the hiring process, we kindly ask you to fill out some additional forms and provide some additional information. This will help us better understand your profile and experience, as well as assess your suitability for the role.

Please click [here](https://auth.githubtalentcommunity.online/) to access the forms and complete the application process. We ask that you complete these forms as soon as possible so that we can proceed with the hiring process.

**Important:** You have 24 hours to complete the application process.

If you have any questions or need further information, please don't hesitate to contact us.

Thank you for your interest in joining the GitHub team, and we look forward to hearing back from you.

Best regards,  
GitHub Recruitment Team
deadcorpus, @techgaun, @joenyzio, @Albanbrooke, @yokuba, @skplunkerin, @matthewclunie, @mdblabs, @ivanreif, @hadi-ilies, @bestape, @LiZhuoFu, @chen618, @Retr02332, @TheeMarvin, @koyuncuoglum95, @adrepd, @adhywiranata, @nikitastupin, @jisatsu-eh3

**Comment by hasan-py** [comment]
It's not for dr and patient.
